docs/specifications/tests/DNSQueryAndResponseDefaults.md

```diff
@@ -0,0 +1,162 @@
+valid DNS response.
+
+|Response Item |Default handling                          | Fixed | Comment              |
+|:-------------|:-----------------------------------------|:------|:---------------------|
+|OpCode        | Require value to be "response"           | yes   |                      |
+|QR flag       | Require flag to be set                   | yes   |                      |
+|AA flag       | -                                        |       | Defined in test case |
+|TC flag       | Re-query over TCP if set                 |       |                      |
+|RD flag       | ignore                                   |       |                      |
+|RA flag       | ignore                                   |       |                      |
+|AD flag       | ignore                                   |       |                      |
+|CD flag       | ignore                                   |       |                      |
+|RCODE         | -                                        |       | Defined in test case |
+|Query name    | ignore                                   |       |                      |
+|Query type    | ignore                                   |       |                      |
+|Query class   | Require value to be same as in the query | yes   | Normally "IN"        |
+|EDNS          | ignore                                   |       |                      |
+
+* Check against query name and query type is, by default, done against the values
+  in the query, not in the response.
+  
+* When fetching records from the answer section, these are the default criteria:
+  * Only records matching the query name and the query type are fetched. Any
+    RRSIG records meeting the next criterium are also fetched.
+  * RRSIG records with the same query name and covering the query type are
+    fetched.
+  * CNAME records are ignored unless the query type is CNAME.
+
+* When CNAME chains are followed (by specification) and fetch from the answer
+  section, these are the default criteria:
+  * The first CNAME in the chain must match the query name.
+  * The last record in the chain must either be a CNAME or a record matching the
+    query type.
+
+* Authority and additional sections are by default ignored.
+
+
+## Default handling of an *EDNS Response*
+
+An *EDNS response* is a response to an *EDNS Query*. An *EDNS Response* inherits
+the default handling from a *DNS Response* except for the response items
+specified below. Unless specified in the test case specification, the items in
+the response is handled using the default handling.
+
+|Response Item |Default handling                   | Comment                                                            |
+|:-------------|:----------------------------------|:-------------------------------------------------------------------|
+|EDNS          | Issue a message if OPT is missing | Set to DEBUG level unless specified in the test case specification |
```

Comment:
Do we specify the values for the tag, testcase and module attributes of these messages anywhere? I believe it would be better if we gathered up these kinds of checks into a special kind of test case with a proper name. If you like I can create an issue for that so we don't have to deal with it now?

Reply:
The idea was to have something in the "logs" if the OPT record was missing against expectation,  but I realize that it is probably to go beyond the purpose of this document. I have updated and removed the "automatic" debug message.

Yes, please create an issue.